platforms/common/uORB/SubscriptionInterval.hpp

```diff
@@ -125,8 +125,13 @@ class SubscriptionInterval
 	{
 		if (_subscription.copy(dst)) {
 			const hrt_abstime now = hrt_absolute_time();
-			// shift last update time forward, but don't let it get further behind than the interval
-			_last_update = math::constrain(_last_update + _interval_us, now - _interval_us, now);
+
+			// make sure we don't set a timestamp before the timer started counting (now - _interval_us would wrap because it's unsigned)
+			if (now > _interval_us) {
+				// shift last update time forward, but don't let it get further behind than the interval
+				_last_update = math::constrain(_last_update + _interval_us, now - _interval_us, now);
+			}
```

Comment:
```suggestion
			} else {
				_last_update = now;
			}
			
			
```

Reply:
Alternatively just change the lower bound of the constrain so it's 0 if (now - _interval_us) will wrap.